sql/wsrep_var.cc

```diff
@@ -256,12 +256,12 @@ static int wsrep_provider_verify (const char* provider_str)
 @return false if no error encountered with check else return true. */
 bool wsrep_provider_check (sys_var *self, THD* thd, set_var* var)
 {
+  char wsrep_provider_buf[FN_REFLEN];
+
   if (WSREP(thd))
   {
-    mysql_mutex_lock(&LOCK_wsrep_pause_count);
-    bool node_paused = (wsrep_pause_count == 0) ? false : true;
+    bool node_paused = thd->global_read_lock.provider_resumed() ? false : true;
```

Comment:
This will take care of pause if fired from same connection only.
(If this stmt is fired from another connection then it will not evaluate node_paused to true.)